websites/#/Фугапедия/presence.ts

```diff
@@ -4,250 +4,172 @@ const presence = new Presence({
+			break;
+		}
+		case "offer": {
+			presenceData.details = "Читает публичную оферту";
 			break;
 		}
-		case "/privacypolicy": {
+		case "privacy": {
 			presenceData.details = "Читает политику";
 			presenceData.state = "конфиденциальности";
-
 			break;
 		}
-		case "/termsofuse": {
-			presenceData.details = "Читает условия";
-			presenceData.state = "использования";
-
+		case "s": {
+			const query = parseName(3);
+			if (!privacyMode && query) {
+				presenceData.details = "Ищет что-то по запросу";
+				presenceData.state = `«${query}»`;
+			} else presenceData.details = "Ищет что-то";
 			break;
 		}
-		case "/ad": {
-			presenceData.details = "Читает условия";
-			presenceData.state = "размещения рекламы";
-
+		case "subscribe": {
+			presenceData.details = "Собирается купить подписку";
 			break;
 		}
-		case "/account": {
-			if (document.location.hash === "#admintools")
-				presenceData.details = "Что-то делает в админ-панели";
-			else presenceData.details = "Редактирует свой аккаунт";
-
+		case "t": {
+			presenceData.details = "Смотрит категорию";
+			if (!privacyMode) presenceData.state = `«${parseName(3)}»`;
 			break;
 		}
-		case "/sub": {
-			presenceData.details = "Собирается купить подписку";
+		case "tags": {
+			presenceData.details = "Смотрит все категории";
 			break;
 		}
-		case "/upload": {
-			presenceData.details = "Загружает изображение";
+		case "termsofuse": {
+			presenceData.details = "Читает условия использования";
 			break;
 		}
-		case "/api.php": {
-			presenceData.details = "Читает документацию API";
+		case "u": {
+			presenceData.details = "Смотрит профиль пользователя";
+			if (!privacyMode) {
+				if (splittedPathname[2] === "me") {
+					presenceData.details = "Смотрит свой профиль";
+					presenceData.state = document.querySelector(
+						"body > div > div > main > div > div > div > div > div > h2.font-serif-text"
+					).textContent;
+				} else presenceData.state = parseName("profile_user_id");
+				presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+					"body > div > div > main > div > div > div > div > div > img"
+				).src;
+				presenceData.smallImageKey = "logo";
+			}
 			break;
 		}
-		default:
-			presenceData.details = "Что-то делает на Фугапедии";
+		default: {
+			if (document.querySelector<HTMLInputElement>("#article_id")) {
+				presenceData.details = "Читает статью";
+				if (!privacyMode) {
+					presenceData.state = `«${parseName(2, "article_id")}»`;
+					const image = document.querySelector<HTMLImageElement>(
+						"body > div > div > div > div > img"
+					);
+					if (image) {
+						presenceData.largeImageKey = image.src;
+						presenceData.smallImageKey = "logo";
+					}
+				}
+			}
+		}
 	}
 
 	presence.setActivity(presenceData);
 });
+
+function parseName(start: number): string;
+function parseName(id: string): string;
+function parseName(start: number, id: string): string;
+function parseName(start: number | string, id?: string): string {
```

Comment:
all these overloads return the same thing, making them useless. You can just remove them and keep the general declaration